packages/ckeditor5-engine/tests/model/utils/deletecontent.js

```diff
@@ -235,19 +237,56 @@ describe( 'DataController utils', () => {
 				'<paragraph>x</paragraph><paragraph>fo[]ar</paragraph><paragraph>y</paragraph>'
 			);
 
+			test(
+				'removes first element when it\'s empty but second element is not empty (same name)',
+				'<paragraph>x</paragraph><paragraph>[foo</paragraph><paragraph>b]ar</paragraph><paragraph>y</paragraph>',
```

Comment:
The test description is a bit misleading and maybe a test case should properly show which node is removed. This would be showcased when using differnt node names.

`<blockA>[foo</blockA><blockB>b]ar</blockB` -> `<blockB>[]ar<blockB>` ?

Also the test name describes some intermediate state. The `<paragraph>foo</paragraph>` is not empty but the test describes it as "empty".